quickwit/quickwit-indexing/src/actors/index_serializer.rs

```diff
@@ -73,14 +73,22 @@ impl Handler<IndexedSplitBatchBuilder> for IndexSerializer {
         batch_builder: IndexedSplitBatchBuilder,
         ctx: &ActorContext<Self>,
     ) -> Result<(), ActorExitStatus> {
-        let splits: Vec<IndexedSplit> = {
-            let _protect = ctx.protect_zone();
-            batch_builder
-                .splits
-                .into_iter()
-                .map(|split_builder| split_builder.finalize())
-                .collect::<Result<_, _>>()?
-        };
+        let mut splits: Vec<IndexedSplit> = Vec::with_capacity(batch_builder.splits.len());
+        for split_builder in batch_builder.splits {
+            // TODO Consider & test removing this protect guard.
+            //
+            // In theory the controlled directory should be sufficient.
+            let _protect_guard = ctx.protect_zone();
+            if let Some(controlled_directory) = &split_builder.controlled_directory_opt {
+                let io_controls = controlled_directory
```

Comment:
You need to add the metric labels: `index_id={index_id}` and `component=index_serializer`

Reply:
done